addons/account/models/mail_message.py

```diff
@@ -108,30 +120,51 @@ def _compute_account_audit_log_partner_id(self):
     def _search_account_audit_log_partner_id(self, operator, value):
         return self._search_audit_log_related_record_id('res.partner', operator, value)
 
+    def _domain_factory(self, model, operator, value):
+        match model:
+            case 'res.company':
+                return [('id', 'in', self.env['account.move.line'].sudo()._search([('company_id.restrictive_audit_trail', '=', True),]).select('company_id'))]
+            case 'account.move':
+                return [('company_id.restrictive_audit_trail', operator, value)]
+            case 'account.account':
+                return [('used', operator, value), ('company_ids.restrictive_audit_trail', operator, value)]
+            case 'account.tax':
+                return [('id', 'in', self.env['account.move.line'].sudo()._search([('tax_line_id', '!=', False), ('company_id.restrictive_audit_trail', '=', True),]).select('tax_line_id'))]
+            case 'res.partner':
+                return [('id', 'in', self.env['account.move.line'].sudo()._search([('partner_id', '!=', False), ('company_id.restrictive_audit_trail', '=', True),]).select('partner_id'))]
+            case _:
+                return NotImplemented
```

Comment:
`NotImplemented` is an internal of Python when it is looking for compatible operators.
You probably want to raise `NotImplementedError` instead (or something else?)